src/renderers/shared/reconciler/ReactReconciler.js

```diff
@@ -65,8 +65,10 @@ var ReactReconciler = {
     internalInstance, nextElement, transaction, context
   ) {
     var prevElement = internalInstance._currentElement;
-
-    if (nextElement === prevElement && nextElement._owner != null) {
+    if (nextElement === prevElement &&
+        nextElement._owner != null
+        // TODO: Shouldn't we need to do this: `&& context === internalInstance._context`
```

Comment:
Ok, good to know.  Can I get a jsfiddle / small test case that demonstrates this so we can add it to the unit tests?

Adding this line appears to cause a unit test to fail (appears to cause React to fail to invoke the proper lifecycle methods).  I *think* it's because if we don't bail out, we invoke `internalInstance.receiveComponent` without first calling the lifecycle methods.  My top/immediate priority was to fix Ryan's usecase in order to unblock him, but I'll sync with @sebmarkbage asap and see if we can get to the bottom of this.